pyccel/ast/internals.py

```diff
@@ -137,13 +137,17 @@ class Slice(Basic):
     >>> Slice(start, stop, step)
     start : stop : step
     """
-    __slots__ = ('_start','_stop','_step')
-    _attribute_nodes = ('_start','_stop','_step')
+    __slots__ = ('_start','_stop','_step', '_slice_type')
+    _attribute_nodes = ('_start','_stop','_step', '_slice_type')
 
-    def __init__(self, start, stop, step = None):
+    Range = LiteralInteger(1)
+    Element = LiteralInteger(0)
+
+    def __init__(self, start, stop, step = None, slice_type = Element):
```

Comment:
Why is the default slice type `Element`?

Reply:
I just had to initialize it with something. Although, I'm not sure if the naming is correct, I put `Range` to represent `[..., :, ...]`  and `Element` for `[..., 2, ...]` for example.

Reply:
The name seems fine to me.

My concern is not the fact that there is an initialisation, but the fact that you initialise with `Element`. I believe `Slice` objects are also created in the semantic stage. With a default value you obviously don't see an error for this, but these slices are then all considered to be `Element`s. This seems wrong to me given that in python a slice is always a `slice_type = Range` and the `Slice`s created in the semantic stage are the equivalent of pure python slices.